### PR TITLE
Some minor refactoring and improvements

### PR DESCRIPTION
**Pull request by ao508**
Signed-off-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

### CODE REVIEW

File: service/src/main/java/org/mskcc/cmo/metadb/service/MetaDbRequestService.java [hunk 1]
```diff
@@ -1,16 +0,0 @@
-package org.mskcc.cmo.metadb.service;
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/MetaDbRequestService.java:1]: class was renamed to `MetadbRequestService` (lowercase D)

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MetaDbRequestServiceImpl.java [hunk 2]
```diff
@@ -1,180 +0,0 @@
-package org.mskcc.cmo.metadb.service.impl;
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MetaDbRequestServiceImpl.java:1]: renamed this class (lowercase D)

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 3]
```diff
@@ -72,14 +83,64 @@ public void run() {
...
+                            // CMO_REQUEST_METADATA_UPDATE topic as well
+                            // regardless of which level the updates occur, the updates still need
+                            // to be persisted to the database
+
+                            if (requestService.requestHasUpdates(existingRequest, request)) {
+                                // if request-level metadata updates exist then persist and publish updates
+                                if (requestService.requestHasMetadataUpdates(existingRequest, request)) {
+                                    // persist request-level metadata updates to database
+                                    // existingRequest.updateRequestMetadata(request)
+                                    // requestService.updateRequestMetadata(existingRequest);
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:112]: request has its metadata updates at this point

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 4]
```diff
@@ -72,14 +83,64 @@ public void run() {
...
+                                    //        CMO_REQUEST_UPDATE_TOPIC,
+                                    //        mapper.writeValueAsString(
+                                    //              existingRequest.getRequestMetadataList()));
+                                }
+
+                                // only publish sample-level metadata history to CMO_SAMPLE_METADATA_UPDATE
+                                //  for samples containing updates in the new request json received
+                                List<MetaDbSample> updatedSamples =
+                                        requestService.getRequestSamplesWithUpdates(request);
+                                if (!updatedSamples.isEmpty()) {
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:128]: samples with updates are identified and persisted to db

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 5]
```diff
@@ -72,14 +83,64 @@ public void run() {
...
+                                    LOG.info("Publishing Sample-level Metadata updates to "
+                                            + "CMO_SAMPLE_METADATA_UPDATE for " + updatedSamples.size()
+                                            + " samples.");
+                                    // maybe pass along to a diff message handler?
+                                    for (MetaDbSample sample : updatedSamples) {
+                                        // persist sample-level metadata updates to database
+                                        // sampleService.updateSampleMetadata(sample)
+                                        //messagingGateway.publish(CMO_SAMPLE_UPDATE_TOPIC,
+                                        //        mapper.writeValueAsString(sample.getSampleMetadataList()));
+                                    }
```
Comment (ao508) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:138]: Need to confirm: Should we still publish the request json RECEIVED to CMO_NEW_REQUEST for the consistency checker?

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java (hunk 3 above)
Comment (ao508) [reply service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:112]: need to add method which will only update the request-level metadata

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 6]
```diff
@@ -98,66 +113,23 @@ public void run() {
...
-                            // 2. If there are request-level metadata updates:
-                            //   - pass request to RequestUpdateMessageHandler
+                            // 2. Pass request to RequestUpdateMessageHandler
                             //   - message handler will persist updates to request metadata to database
                             //     and will handle the request metadata versioning
                             //   - message handler will also publish request metadata history to
                             //     CMO_REQUEST_METADATA_UPDATE topic
-                            // 3. If there are sample-level metadata updates:
-                            //   - pass request to SampleUpdateMessageHandler
+                            // 3. Pass SampleMetadata to SampleUpdateMessageHandler
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:121]: TODO: handle new SampleMetadata

Comment (divyamadala30) [reply service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:121]: New Samples will be persisted to NEO4J in the SampleMetadataUpdateHandler.

File: service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java [hunk 7]
```diff
@@ -176,11 +148,108 @@ public void run() {
...
+        }
+
+        @Override
+        public void run() {
+            phaser.arrive();
+            while (true) {
+                try {
+                    SampleMetadata sampleMetadata = sampleUpdateQueue.poll(100, TimeUnit.MILLISECONDS);
+                    if (sampleMetadata != null) {
+                        MetaDbSample existingSample = sampleService.getMetaDbSampleByRequestAndIgoId(
```
Comment (divyamadala30) [inline service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:216]: Check if existingSample is null for handling new samples

Comment (divyamadala30) [reply service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java:216]: Note: If the request is not new, any updates to the request metadata or sample metadata will published to their respective yellow channels

**Review by divyamadala30 — COMMENTED** [review]
TODO list:
 - Remove requestJson update from `updateRequestMetadata(MetaDbRequest updatedRequest)` in MetadbRequest.java

**Pull request merged**